File: Chapter7/Python/Example5.py
# Chapter 7 - Τυπική Κανονική Κατανομή
# Example 5: Πλήρης Επιχειρηματική Ανάλυση Πωλήσεων
#   X ~ N(50.000, 8.000²)

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm

# Parameters
mu    = 50000
sigma = 8000

# 1. Probabilities for different ranges
print("=== Sales Analysis Report ===\n")

# Below average
prob_below_avg = norm.cdf(mu, loc=mu, scale=sigma)
print(f"1. Probability of below-average sales: {prob_below_avg*100:.2f}%")

# Above 60000
prob_above_60k = 1 - norm.cdf(60000, loc=mu, scale=sigma)
print(f"2. Probability of sales > 60,000: {prob_above_60k*100:.2f}%")

# Between 45000 and 55000
prob_middle = norm.cdf(55000, loc=mu, scale=sigma) - \
              norm.cdf(45000, loc=mu, scale=sigma)
print(f"3. Probability of sales between 45K-55K: {prob_middle*100:.2f}%")

# 2. Key quantiles
print("\n=== Performance Quantiles ===")
quantiles = [0.1, 0.25, 0.5, 0.75, 0.9, 0.95]
values    = norm.ppf(quantiles, loc=mu, scale=sigma)

for q, v in zip(quantiles, values):
    print(f"{int(q*100)}th percentile: EUR {v:.0f}")

# 3. Visualization
x = np.linspace(mu - 4*sigma, mu + 4*sigma, 1000)
y = norm.pdf(x, loc=mu, scale=sigma)

plt.figure(figsize=(14, 8))
plt.plot(x, y, 'b-', linewidth=2, label='Distribution')

# Shade regions
x_low = x[x <= 45000]
plt.fill_between(x_low, norm.pdf(x_low, loc=mu, scale=sigma),
                 alpha=0.3, color='red', label='Below Target (<45K)')

x_mid = x[(x >= 45000) & (x <= 55000)]
plt.fill_between(x_mid, norm.pdf(x_mid, loc=mu, scale=sigma),
                 alpha=0.3, color='green', label='Target Range (45K-55K)')

x_high = x[x >= 60000]
plt.fill_between(x_high, norm.pdf(x_high, loc=mu, scale=sigma),
                 alpha=0.3, color='blue', label='Excellent (>60K)')

# Add reference lines
plt.axvline(mu,    color='black',   linestyle='--', linewidth=2,
            label=f'Mean = EUR {mu:,.0f}')
plt.axvline(45000, color='darkgray', linestyle=':', linewidth=1.5)
plt.axvline(55000, color='darkgray', linestyle=':', linewidth=1.5)
plt.axvline(60000, color='darkgray', linestyle=':', linewidth=1.5)

# Labels
plt.text(mu,    max(y)*1.05, f'Mean = EUR {mu:,.0f}',
         ha='center', fontsize=10, weight='bold')
plt.text(40000, max(y)*0.5, 'Below\nTarget',
         ha='center', color='red', fontsize=9)
plt.text(50000, max(y)*0.3, 'Target\nRange',
         ha='center', color='darkgreen', fontsize=9)
plt.text(65000, max(y)*0.5, 'Excellent',
         ha='center', color='blue', fontsize=9)

plt.xlabel('Sales (EUR)', fontsize=12)
plt.ylabel('Density', fontsize=12)
plt.title('Sales Distribution Analysis', fontsize=14, weight='bold')
plt.legend(fontsize=10, loc='upper right')
plt.grid(alpha=0.3)
plt.tight_layout()
plt.show()

# 4. Summary statistics table
print("\n=== Summary Statistics ===")
print(f"Mean: EUR {mu:,.0f}")
print(f"Standard Deviation: EUR {sigma:,.0f}")
print(f"Coefficient of Variation: {(sigma/mu)*100:.2f}%")
print(f"\nZ-scores for key values:")
print(f"Sales = 45,000: z = {(45000-mu)/sigma:.2f}")
print(f"Sales = 55,000: z = {(55000-mu)/sigma:.2f}")
print(f"Sales = 60,000: z = {(60000-mu)/sigma:.2f}")

File: Chapter7/R/Example5.R
# Chapter 7 - Τυπική Κανονική Κατανομή
# Example 5: Πλήρης Επιχειρηματική Ανάλυση Πωλήσεων
#   X ~ N(50.000, 8.000²)

# Parameters
mu    <- 50000
sigma <- 8000

# 1. Probabilities for different ranges
cat("=== Sales Analysis Report ===\n\n")

# Below average
prob_below_avg <- pnorm(mu, mean = mu, sd = sigma)
cat(sprintf("1. Probability of below-average sales: %.2f%%\n",
            prob_below_avg * 100))

# Above 60000
prob_above_60k <- 1 - pnorm(60000, mean = mu, sd = sigma)
cat(sprintf("2. Probability of sales > 60,000: %.2f%%\n",
            prob_above_60k * 100))

# Between 45000 and 55000
prob_middle <- pnorm(55000, mean = mu, sd = sigma) -
  pnorm(45000, mean = mu, sd = sigma)
cat(sprintf("3. Probability of sales between 45K-55K: %.2f%%\n",
            prob_middle * 100))

# 2. Key quantiles
cat("\n=== Performance Quantiles ===\n")
quantiles <- c(0.1, 0.25, 0.5, 0.75, 0.9, 0.95)
values    <- qnorm(quantiles, mean = mu, sd = sigma)

for (i in 1:length(quantiles)) {
  cat(sprintf("%dth percentile: EUR %.0f\n",
              quantiles[i] * 100, values[i]))
}

# 3. Visualization
x <- seq(mu - 4*sigma, mu + 4*sigma, length.out = 1000)
y <- dnorm(x, mean = mu, sd = sigma)

plot(x, y, type = 'l', lwd = 2, col = 'blue',
     main = 'Sales Distribution Analysis',
     xlab = 'Sales (EUR)', ylab = 'Density')

# Shade regions
# Below target (45000)
x_low <- x[x <= 45000]
y_low <- dnorm(x_low, mean = mu, sd = sigma)
polygon(c(min(x), x_low, 45000), c(0, y_low, 0),
        col = rgb(1, 0, 0, 0.3), border = NA)

# Target range (45000-55000)
x_mid <- x[x >= 45000 & x <= 55000]
y_mid <- dnorm(x_mid, mean = mu, sd = sigma)
polygon(c(45000, x_mid, 55000), c(0, y_mid, 0),
        col = rgb(0, 1, 0, 0.3), border = NA)

# Excellent (above 60000)
x_high <- x[x >= 60000]
y_high <- dnorm(x_high, mean = mu, sd = sigma)
polygon(c(60000, x_high, max(x)), c(0, y_high, 0),
        col = rgb(0, 0, 1, 0.3), border = NA)

# Add reference lines
abline(v = mu, col = 'black', lwd = 2, lty = 2)
abline(v = c(45000, 55000, 60000),
       col = 'darkgray', lwd = 1.5, lty = 3)

# Labels
text(mu,    max(y) * 1.05,
     sprintf('Mean = EUR %.0f', mu), col = 'black', cex = 0.9)
text(40000, max(y) * 0.5, 'Below\nTarget',     col = 'red',       cex = 0.8)
text(50000, max(y) * 0.3, 'Target\nRange',     col = 'darkgreen', cex = 0.8)
text(65000, max(y) * 0.5, 'Excellent',         col = 'blue',      cex = 0.8)

legend('topright',
       legend = c('Below Target (<45K)',
                  'Target Range (45K-55K)',
                  'Excellent (>60K)'),
       fill   = c(rgb(1,0,0,0.3), rgb(0,1,0,0.3), rgb(0,0,1,0.3)),
       cex    = 0.8)

grid()
